### DISCUSSION TITLE
Z-Offset still unreliable, so it seems

### DISCUSSION THREAD

**Discussion by Miq1** [Jyers/Marlin#1915] [Q&A]
I am afraid I have to open this again, I hope someone has a clue what might have happened.

I am using a 2.0.2i-RET6 version, compiled with the Maple environment, since I have one of the newer 4.2.2 boards with the GD32 MCU on it. Manual mesh, by the way.

After having set up printer and mesh from scratch again some days ago, I successfull printed several PETG parts without a flaw. I had a Z-Offset of 0.1mm, since the paper sheet I used to level is 0.1mm thick. After some days I changed filaments and started printing TPU now. Again for two days, everything was fine, with unchanged settings.

Today I switched back to PETG. Out of a sudden, nothing would stick any more to the build plate, despite intense cleaning and unchanged settings. So I started again leveling and found the print bed obviously about 1-2 tenths too low. Strange, since I did nothing to it and no one else could have touched the printer.

So I did the leveling procedure again as before and created a fresh mesh as well. The mesh did not differ much, by the way, just by some 1/100th at a few points. Then set the Z-Offset to -0.1 again and started printing.

A mess... After several attempts I found that the Z-Offset was much too deep, so the nozzle would scrape the print from the bed. With a setting of 0 I now am able to print.

But why? A potential explanation could be the Z-Offset was not considered when the print bed was claimed to have sunk, but restored afterwards without me knowing. I am pretty sure, though, to have set and stored it as ever, and did not have touched it at all when the problem popped up. The display readings were normal, the Z-Offset value always was as I had expected.

Funny enough that this only happened today, after having worked for several days before.

**Comment by Miq1** [comment]
It happened again. After being able to print flawlessly for a couple of days, today the print bed seems to have fallen down again by at least 1-2 tenths, in between two prints. I know that cannot happen, so it must have to do with the firmware. There are no Cura settings or start code involved (no other than the print before), the print file is identical, the temperatures are, the printer was left running.

I had to redo manual leveling and setting up a fresh mesh before I was able to print again.

This happened to me four or five times now. If only I could make out a pattern...

**Comment by BlueWings172** [comment]
@Miq1 

I have experienced issues with z offset changing and I have read posts from other people having the same problem. For me, all this stopped once I reverted back to version 1.3.5b.

**Comment by Miq1** [comment]
May be I do not get it right yet: the Z offset issue seems to have hit users of the base Marlin FW as well, but reportedly has been fixed in the meantime with the 2.0.9.3 bugfix code, if compiled for the STM32F103 MCU with the Maple environment.
So if the jyers code is based on 2.0.9.3 bugfix, the issue should have vanished in it as well - is it?

Else, I could try to get the basic Marlin running on my Ender 3 V2, but would lose the benefits of the Jyers UI improvements, right?

**Reply by Miq1** [reply]
Thank you for the explanations. You are right, I did not look into the code yet to see if there is a chance to use f.i. an uninitialized variable at certain circumstances. Get me right: I am not blaming you to be responsible at all, I was hoping you will know the Marlin code excellently and perhaps could think of a way this may happen.

I have difficulties, though, to see which parts of the hardware could be causing this. 
- The Z endstop is a microswitch that seems to work - it switches and connects the wires as supposed to do. 
- The Z steppers (2) obviously are working most of the time while printing stuff without any observable faults in the print. 
- The bed springs (yes, I switched from the silicone pads to yellow hard springs in the meantime to sort out those as a cause) and leveling knobs are working linearly as expected, there is no reason why all 4 should go down (against the spring force!) simultaneously. 
- The steel plate, adhesive pad and aluminium bed will not jump either.
- If it was a hardware effect, I would expect it to happen more often, not only once in a while after days of correct function.

**Reply by Miq1** [reply]
> @Miq1 I advise you to look at our Discord server. A new build is available. And note : with Manual Mesh, the bed leveling for 4 corners (bed tramming) must be do, before Z-Offset and before create mesh.

Of course, that is what I am doing.

I have never been on Discord before. Will try to see if I can get along.

**Reply by Miq1** [reply]
Uhm - where may I find the latest build? All places I found so far seem to be older versions...

I am on Discord now as Miq19, BTW.
